### DISCUSSION TITLE
Full string match

### DISCUSSION THREAD

**Discussion by XakepSDK** [rust-lang/regex#737] [Q&A] [answered]
In java, i could do `"string".matches("regexStr")` and it will check if whole string matches to a given regex.
I haven't found a way to do this with this crate.
I'm not sure if i should open feature request or it's already possible, just i haven't noticed the way

**Comment by BurntSushi** [comment, marked as answer]
From [the docs](https://docs.rs/regex/1.4.2/regex/struct.Regex.html):

> All searching is done with an implicit `.*?` at the beginning and end of an expression. To force an expression to match the whole string (or a prefix or a suffix), you must use an anchor like `^` or `$` (or `\A` and `\z`).

So in your case, you would want:

```rust
use regex::Regex;

fn main() {
    let matched = Regex::new(r"^regexStr$").unwrap().is_match("string");
    println!("{:?}", matched);
}
```

**Reply by XakepSDK** [reply]
Regex comes over the network. Is it safe to modify incoming regex this way? May this bring some unforeseen consequences?

**Reply by BurntSushi** [reply]
If you already trust the regex itself, then sure, I don't see why not. `^(?:regex-from-network)$` should do the trick. Now, `regex-from-network` could break out of the parenthetical and subvert your approach, but only by crafting an invalid regex. So you could check that `Regex::new(regex-from-network)` is itself a valid regex first. But if you're at that point, then it means your regex is untrusted itself, which could pose other problems. (Performance wise.)

**Comment by 35VLG84** [comment]
I had this same problem with one of my projects (JDK vs. Rust regex match logic difference) 

The regex was part of JSON serialized interface, and also part of the API was a checksum calculated over the Regex Pattern. So fixing this difference in semantics was little bit tricky. 

The code is here, and it's licensed as Apache License, version 2.0 or MIT, at your option:
https://github.com/e257-fi/tackler-ng/commit/1f5a95030a1dedbd400ed79c85df98dffd00d54d

Example how this is used: 
https://github.com/e257-fi/tackler-ng/commit/407425aecefdf7c6389d6dcc65af0d156d9c4439
